### PR TITLE
APP-510: Extend apptester.py and related test suits to all the applications coverage

### PR DESCRIPTION
**Pull request by mariangavlytskiy**
1) Extended kaautils.py with new REST methods, and added possibility to check build of source.
2) Modified apptester.py (added validation of source build for all application which have it.)

### CODE REVIEW

**Comment by kaa-leeroy** [comment]
Can one of the admins verify this patch?

File: scripts/apptester/apptester.py [hunk 1]
```diff
@@ -164,8 +164,9 @@ def __init__(self, config_file, kaanode, kaauser, rootpath, testdir, sandboxfram
 
         # list of applications that must be skipped during building ant testing
         self.skipped = []
-
         self.applications = self._create_applications(config_file)
+        self.BINARY = 'BINARY'
```
Comment (vadimol) [inline scripts/apptester/apptester.py:168]: These properties are not logically connected with AppTesterFramework. Consider to move them to a separate class (for example AppFormat). The format is a property of the application.

File: scripts/apptester/apptester.py [hunk 2]
```diff
@@ -251,33 +252,41 @@ def build_applications(self, name=None):
             build_result = [app_data, self.result_matrix[app], 'N/A']
             self.table_data.append(build_result)
 
-    def build_android_java_demo(self):
+
+    def execute_build(self, app_data, file_type):
```
Comment (vadimol) [inline scripts/apptester/apptester.py:256]: I would suggest renaming this method to "execute_remote_build" or something like this.

File: scripts/apptester/apptester.py [hunk 3]
```diff
@@ -251,33 +252,41 @@ def build_applications(self, name=None):
...
-                    
-                    except Exception as ex:
-                        traceback.print_exc(ex)
-                        self.result_matrix[item['name']] = TestStatus.FAILED
+                    self.execute_build(item, self.BINARY)
 
-        for app in self.result_matrix:
-            build_result = [app, self.result_matrix[app], 'N/A']
-            self.table_data.append(build_result)
+    def build_all_demo(self):
```
Comment (vadimol) [inline scripts/apptester/apptester.py:284]: This method actually does not build all of the demos. Based on code bellow, I would recommend to rename it to "check_src_archives" or "verify_src_archives"

File: scripts/apptester/apptester.py [hunk 4]
```diff
@@ -287,6 +296,9 @@ def process_results(self, output=False):
         passed = True
 
         # TODO APP-53 Add test results
+        for app in self.result_matrix:
```
Comment (vadimol) [inline scripts/apptester/apptester.py:299]: Is this related to APP-510?

File: scripts/apptester/apptester.py [hunk 5]
```diff
@@ -310,6 +322,8 @@ def console_args_parser():
                         help='specify application')
     parser.add_argument('-j', help='build java/android applications',
                         action='store_true')
+    parser.add_argument('-c', help='build all source applications.',
```
Comment (vadimol) [inline scripts/apptester/apptester.py:325]: This must be tested on Jenkins

File: scripts/apptester/kaautils.py [hunk 6]
```diff
@@ -137,6 +139,103 @@ def get_sdk_profiles(self, appname, kaauser):
 
         return req.json()
 
+    def create_configuration(self, kaauser, configuration):
```
Comment (vadimol) [inline scripts/apptester/kaautils.py:142]: Are these changes related to APP-510?

File: scripts/apptester/apptester.py [hunk 7]
```diff
@@ -364,7 +378,9 @@ def main():
     tester = AppTesterFramework(appconfig_file, kaanode, kaauser,
                                 args.rootpath, builddir, sandboxframe)
     if args.j:
-        tester.build_android_java_demo()
+        tester.build_java_android_demo()
```
Comment (vadimol) [inline scripts/apptester/apptester.py:381]: We need to add additional phase on Jenkins that will run the apptester with "-c" option.

**Review by vadimol — CHANGES_REQUESTED** [review]
See comments. Overall, looks good.

File: scripts/apptester/apptester.py (hunk 4 above)
Comment (mariangavlytskiy) [reply scripts/apptester/apptester.py:299]: Yes, it was made for simplify result processing after adding new test.

File: scripts/apptester/kaautils.py (hunk 6 above)
Comment (mariangavlytskiy) [reply scripts/apptester/kaautils.py:142]: No, these methods were made before for future testing of applications.

**Review by Slash32 — CHANGES_REQUESTED** [review]
Please re-base all the commits onto master avoiding merge commits.

**Comment by Slash32** [comment]
@kaa-leeroy add to whitelist

**Pull request closed without merging**